File: backend/src/main/java/com/backend/domain/user/entity/RefreshToken.java
package com.backend.domain.user.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.redis.core.RedisHash;
import org.springframework.data.redis.core.TimeToLive;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Entity
@Builder
@RedisHash(value = "refreshToken")
public class RefreshToken {

    @Id
    private Long userId;

    private String refreshToken;

    @TimeToLive
    private Long expiration;
}

File: backend/src/main/java/com/backend/domain/user/entity/RefreshToken.kt
package com.backend.domain.user.entity

import jakarta.persistence.Entity
import jakarta.persistence.Id
import org.springframework.data.redis.core.RedisHash
import org.springframework.data.redis.core.TimeToLive

@Entity
@RedisHash(value = "refreshToken")
class RefreshToken(

    @Id
    val userId: Long,

    var refreshToken: String,

    @TimeToLive
    var expiration: Long

)
